src/alloc.f90

```diff
@@ -217,7 +438,7 @@ subroutine alloc_z_real(var, decomp, opt_global)
      if (present(opt_global)) then
         global = opt_global
      else
-        global = .false.
+        global = default_opt_global
```

Comment:
This is potentially a confusing or significant change from how the code works now - would it be more appropriate to do in v3? (v2.1 shouldn't change behaviour or ideally interface over v2).

Reply:
The parameter `default_opt_global` is set to false in the PR (inside decomp_2d.f90). My intention is to have only a single parameter for all the alloc subroutines instead of hard-coded values duplicated

Reply:
As a side remark, the new design will change the IO interface.